### ISSUE TITLE
UserWarning: None of the inputs have requires_grad=True. Gradients will be None

### ISSUE THREAD

**Issue by TUDelftHao** [tulerfeng/Video-R1#50]
请问训练开始前打印了这样的告警对训练会有影响吗？

<img width="1161" alt="Image" src="https://github.com/user-attachments/assets/a59f0f50-0373-4b7d-812a-6729f0d1eb87" />

训练过程的loss和reward曲线如下：

<img width="604" alt="Image" src="https://github.com/user-attachments/assets/c99a59ce-2de6-48bd-998c-12391645e432" />

接触强化学习不多，感谢赐教！🙏

**Comment by tulerfeng** [comment]
您好：

1. 打印出这个是正常的，没问题
2. 曲线也没问题，loss项主要由两部分构成：advantages和KL散度，其中advantages由于GRPO的归一化基本接近于0，所以主要是KL散度，随着训练的进行，模型的policy会逐渐偏离原来的分布，所以loss增加是正常的

**Comment by TUDelftHao** [comment]
> 您好：
> 
> 1. 打印出这个是正常的，没问题
> 2. 曲线也没问题，loss项主要由两部分构成：advantages和KL散度，其中advantages由于GRPO的归一化基本接近于0，所以主要是KL散度，随着训练的进行，模型的policy会逐渐偏离原来的分布，所以loss增加是正常的

好的，感谢解答！🙏

**Comment by buchuoer** [comment]
> 您好：
> 
> 1. 打印出这个是正常的，没问题
> 2. 曲线也没问题，loss项主要由两部分构成：advantages和KL散度，其中advantages由于GRPO的归一化基本接近于0，所以主要是KL散度，随着训练的进行，模型的policy会逐渐偏离原来的分布，所以loss增加是正常的

这样的设计，advantages是不是就是多余的？最终时序性奖励、准确率等差异无法在loss中体现出来

**Comment by tulerfeng** [comment]
Hi，这样advantages的设计是follow了Deepseek-R1，虽然准确率等没法在loss曲线中反应出来，但是可以通过梯度反向传播去优化模型，所以是没问题的。同时，准确率等一系列rewards也单独打印了出来，方便观察模型训练。当然，未来工作也欢迎社区尝试提出更好的RL训练算法去训练模型。
